server/api/db/sqldb/db.py

```diff
@@ -4353,6 +4353,10 @@ def get_alert_template(
             self._get_alert_template_record(session, name)
         )
 
+    def get_all_alerts(self, session) -> list[mlrun.common.schemas.AlertConfig]:
+        query = self._query(session, AlertConfig)
+        return list(map(self._transform_alert_config_record_to_schema, query.all()))
```

Comment:
since you are retrieving all db records and assume some alerts configs are bound to some heuristic amount of records, please make the query limited by that calculation (or any config value) - eg 1000 records
that way, you are protecting mlrun from overflowing its memory